Update README and Dockerfile

juliu-s opened the pull request:
As we discussed I created this pr with small enhancements and updated the readme.

Commit:
update readme

Commit:
update dockerfile

Commit:
Merge branch 'patch-1'

trampgeek commented:
Many thanks Julius. I've merged your changes and made a couple of other 
minor tweaks to the documentation. I also updated to openjdk-14 (was 
11). All now pushed to github.

The only thing I didn't fully understand was the inclusion of /apt 
install acl/. That can't do any harm, but is it not installed in the 
base ubuntu image anyway? However, it's good documentation, anyway - I 
recall some other Linux flavours didn't have acl included. Did you have 
some particular issue that caused you to add that?

Regarding home directories (earlier correspondence) I hadn't noticed 
that /etc/passwd had home directories for the jobe users. It seems the 
/useradd/ command does that, even when you tell it not to create home 
directories. Strange. But it doesn't seem to cause any issues.

I'm sure you're right that one /could/ construct home directories for 
all the jobe users and make them secure but my take on it is that it's 
just one more potential attack vector. As I see it, home directories are 
where users store personal configuration and other info, which I just 
don't wish to allow in Jobe. So far at least there hasn't been an 
application that has absolutely required the existence of a home 
directory, though a few have needed a bit of coaxing to play along.

Thanks again for the continued support.

Regards

Richard

On 1/04/21 10:06 pm, Julius wrote:
>
> As we discussed I created this pr with small enhancements and updated 
> the readme.
>
> ------------------------------------------------------------------------
>
>
>         You can view, comment on, or merge this pull request online at:
>
> https://github.com/trampgeek/jobeinabox/pull/6
>
>
>         Commit Summary
>
>   * update readme
>   * update dockerfile
>   * Merge branch 'patch-1'
>
>
>         File Changes
>
>   * *M* Dockerfile
>     <https://github.com/trampgeek/jobeinabox/pull/6/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557>
>     (67)
>   * *M* README.md
>     <https://github.com/trampgeek/jobeinabox/pull/6/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>     (15)
>
>
>         Patch Links:
>
>   * https://github.com/trampgeek/jobeinabox/pull/6.patch
>   * https://github.com/trampgeek/jobeinabox/pull/6.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/trampgeek/jobeinabox/pull/6>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAIM2VT25Z275TPFK7VLJLDTGQZQZANCNFSM42GWEO5Q>.
>

juliu-s commented:
Hi Richard,

The apt install acl was already there[1], I just moved every package to a newline for an better overview.

Regarding the home directories, I am not sure how the webserver (www-data) and the jobe application with the jobe user(s) interacts / are related with eachother. I see that the user www-data (that runs the jobe application) has write access to /home/jobe/files and /home/jobe/runs. But the jobe01, 02 etc users don't.

docker run --rm -it docker.io/trampgeek/jobeinabox bash
***@***.***:/# su -s /bin/bash -c "touch /home/jobe/files/foobar" www-data
***@***.***:/# ls -l /home/jobe/files/
total 0
-rw-rw-r-- 1 www-data www-data 0 Apr 6 18:52 foobar
***@***.***:/# su -s /bin/bash -c "touch /home/jobe/files/barfoo" jobe
***@***.***:/# ls -l /home/jobe/files/
total 0
-rw-rw-r-- 1 jobe jobe 0 Apr 6 18:52 barfoo
-rw-rw-r-- 1 www-data www-data 0 Apr 6 18:52 foobar
***@***.***:/# su -s /bin/bash -c "touch /home/jobe/files/barfoobarfoo" jobe01
touch: cannot touch '/home/jobe/files/barfoobarfoo': Permission denied

In a default environment users don't have write access to other users homedirs, but all the jobe users are member of the same jobe group. So if there is group write access on a directory with the group ownership jobe they could exchange files with eachother. But that is not the case for /home/jobe. For now the homedirs for jobe01, 02 etc are not created and there is no group write access on /home/jobe, so when you change user to jobe01 or jobe02 (the jobe users don't have a shell, but you can work arround that if you are root with "su -s /bin/bash jobe01") you can't do much.

With the useradd commando you can add the -m option to actual create the homedir, or use -d /dev/null, or you can do something like this:

$ docker run --rm -it docker.io/ubuntu:20.04 bash
***@***.***:/# groupadd jobe
***@***.***:/# mkdir /home/shared_jobe
***@***.***:/# chmod 770 /home/shared_jobe
***@***.***:/# chown root:jobe /home/shared_jobe
***@***.***:/# useradd -g jobe -d /home/shared_jobe jobe01
***@***.***:/# useradd -g jobe -d /home/shared_jobe jobe02
***@***.***:/# su - jobe01
$ pwd
/home/shared_jobe
$ touch jobe01_file1
$ exit
***@***.***:/# su - jobe02
$ touch jobe02_file2
$ ls -l
total 0
-rw-r--r-- 1 jobe01 jobe 0 Apr 6 06:41 jobe01_file1
-rw-r--r-- 1 jobe02 jobe 0 Apr 6 06:41 jobe02_file2
$ pwd
/home/shared_jobe

I hope it's not to confusing and it helps a bit.

Best regards, Julius

[1] https://github.com/trampgeek/jobeinabox/blob/420924ee18dbcddc75320f42aae7a48bee23cc60/Dockerfile

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
On Saturday, April 3, 2021 7:41 AM, Richard Lobb ***@***.***> wrote:

> Many thanks Julius. I've merged your changes and made a couple of other
> minor tweaks to the documentation. I also updated to openjdk-14 (was
> 11). All now pushed to github.
>
> The only thing I didn't fully understand was the inclusion of /apt
> install acl/. That can't do any harm, but is it not installed in the
> base ubuntu image anyway? However, it's good documentation, anyway - I
> recall some other Linux flavours didn't have acl included. Did you have
> some particular issue that caused you to add that?
>
> Regarding home directories (earlier correspondence) I hadn't noticed
> that /etc/passwd had home directories for the jobe users. It seems the
> /useradd/ command does that, even when you tell it not to create home
> directories. Strange. But it doesn't seem to cause any issues.
>
> I'm sure you're right that one /could/ construct home directories for
> all the jobe users and make them secure but my take on it is that it's
> just one more potential attack vector. As I see it, home directories are
> where users store personal configuration and other info, which I just
> don't wish to allow in Jobe. So far at least there hasn't been an
> application that has absolutely required the existence of a home
> directory, though a few have needed a bit of coaxing to play along.
>
> Thanks again for the continued support.
>
> Regards
>
> Richard
>
> On 1/04/21 10:06 pm, Julius wrote:
>>
>> As we discussed I created this pr with small enhancements and updated
>> the readme.
>>
>> ------------------------------------------------------------------------
>>
>>
>> You can view, comment on, or merge this pull request online at:
>>
>> https://github.com/trampgeek/jobeinabox/pull/6
>>
>>
>> Commit Summary
>>
>> * update readme
>> * update dockerfile
>> * Merge branch 'patch-1'
>>
>>
>> File Changes
>>
>> * *M* Dockerfile
>> <https://github.com/trampgeek/jobeinabox/pull/6/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557>
>> (67)
>> * *M* README.md
>> <https://github.com/trampgeek/jobeinabox/pull/6/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>> (15)
>>
>>
>> Patch Links:
>>
>> * https://github.com/trampgeek/jobeinabox/pull/6.patch
>> * https://github.com/trampgeek/jobeinabox/pull/6.diff
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/trampgeek/jobeinabox/pull/6>, or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAIM2VT25Z275TPFK7VLJLDTGQZQZANCNFSM42GWEO5Q>.
>>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, [view it on GitHub](https://github.com/trampgeek/jobeinabox/pull/6#issuecomment-812816227), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AENRNJLMR2OZYNBY3VLSF4TTG2TBLANCNFSM42GWEO5Q).
